CUDADataFormats/Track/interface/TrackSoAHeterogeneousT.h

```diff
@@ -42,8 +43,27 @@ class TrackSoAHeterogeneousT {
   // this is chi2/ndof as not necessarely all hits are used in the fit
   eigenSoA::ScalarSoA<float, S> chi2;
 
+  eigenSoA::ScalarSoA<int8_t, S> nLayers;
+
   constexpr int nHits(int i) const { return detIndices.size(i); }
 
+  constexpr bool isTriplet(int i) const { return nLayers(i) == 3; }
+
+  constexpr int computeNumberOfLayers(int32_t i) const {
+    // layers are in order and we assume tracks are either forward or backward
+    auto pdet = detIndices.begin(i);
+    int nl = 1;
+    auto ol = phase1PixelTopology::getLayer(*pdet);
+    for (; pdet < detIndices.end(i); ++pdet) {
+      auto il = phase1PixelTopology::getLayer(*pdet);  // , ol);
+      if (il != ol)
+        ++nl;
+      ol = il;
+    }
+    assert(nl >= 3);
```

Comment:
why the assert ?
I understand that tracks _should_ have at least 3 hits, but it's not a requirement of the code, is it ?

Reply:
formally You are right. was just there to catch a possible bug. will delete.